tracer.go

```diff
@@ -23,4 +21,8 @@ type Tracer interface {
 	//         SetTag("lucky_number", 42)
 	//
 	StartTrace(operationName string) Span
+
+	// Return a WireExtractor for the given WireEncoding, or nil if the Tracer
+	// does not support such an encoding.
+	WireExtractorForEncoding(enc WireEncoding) WireExtractor
```

Comment:
extraction, of course, needs to be on the Tracer, not on the Span (since there *is* no Span at this point)